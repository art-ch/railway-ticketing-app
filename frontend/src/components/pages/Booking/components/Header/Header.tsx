import React from 'react';

export type HeaderProps = { bookingId?: string };

export const Header = ({ bookingId }: HeaderProps) => (
  <div>
    <p className="absolute left-4 top-4 border-2 border-black px-1">22</p>
    <div className="flex gap-2 items-baseline justify-center font-bold pb-2">
      <p className="text-sm">art-ch railway</p>
      <p className="text-lg border-l border-r border-black px-2">
        Travel Document
      </p>
      {!bookingId && <p className="text-sm px-2 text-gray-300">booking id</p>}
      {bookingId && (
        <p className="text-sm text-nowrap w-33">
          {/** shortening ID for demo purposes */}
          ID #{bookingId.replace(/(.*?)-.*/, '$1')}
        </p>
      )}
    </div>
  </div>
);
