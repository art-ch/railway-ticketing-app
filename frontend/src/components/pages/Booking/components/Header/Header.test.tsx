import { render } from '@testing-library/react';
import { Header } from './Header';

describe('Header component', () => {
  it('should render correctly', () => {
    const { container } = render(<Header bookingId={'1'} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly without bookingid', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
