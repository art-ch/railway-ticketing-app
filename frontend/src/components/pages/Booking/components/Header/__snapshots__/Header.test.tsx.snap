// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div>
  <div>
    <p
      class="absolute left-4 top-4 border-2 border-black px-1"
    >
      22
    </p>
    <div
      class="flex gap-2 items-baseline justify-center font-bold pb-2"
    >
      <p
        class="text-sm"
      >
        art-ch railway
      </p>
      <p
        class="text-lg border-l border-r border-black px-2"
      >
        Travel Document
      </p>
      <p
        class="text-sm text-nowrap w-33"
      >
        ID #
        1
      </p>
    </div>
  </div>
</div>
`;

exports[`Header component should render correctly without bookingid 1`] = `
<div>
  <div>
    <p
      class="absolute left-4 top-4 border-2 border-black px-1"
    >
      22
    </p>
    <div
      class="flex gap-2 items-baseline justify-center font-bold pb-2"
    >
      <p
        class="text-sm"
      >
        art-ch railway
      </p>
      <p
        class="text-lg border-l border-r border-black px-2"
      >
        Travel Document
      </p>
      <p
        class="text-sm px-2 text-gray-300"
      >
        booking id
      </p>
    </div>
  </div>
</div>
`;
